routes/api.js

```diff
@@ -2,17 +2,77 @@ const express = require( "express" );
 const router = express.Router();
 const bcrypt = require( "bcrypt" );
 const passport = require( "passport" );
+const methodOverride = require( "method-override" );
+
+const { checkNotAuthenticated, checkAuthenticated, returnAuthentication } = require( "../util/passportUtil" );
 
-const passport_config = require( "../util/passportUtil" );
-const { checkNotAuthenticated } = passport_config;
 const randomColor = require( "../util/randomColor" );
 
+router.use( returnAuthentication );
+router.use( methodOverride( "_method" ) );
+
 router.post( "/login", passport.authenticate( "local", {
-  successRedirect: "/home",
+  successRedirect: "/app",
   failureRedirect: "/login",
   failureFlash   : true,
 } ) );
 
+router.put( "/user/password", checkAuthenticated, async ( req, res, next ) => {
```

Comment:
Verstehe ich das richtig, dass du hier eine Route und Action für Passwort- (`/user/password`) und eine für Namens-Änderung (`/user/username`) gemacht hast?

Nach REST müsste es nur eine Route geben: `PUT /users`.
Auch `DELETE` müsste auf `/users` gehen. Da nach REST du eine Ressource verwaltest und die ist immer in Mehrzahl, also zählbar.

Reply:
Ich habe jetzt alle dies Anfragen unter /users zusammengefasst